proposed/2023/Add-dotnet-search.md

```diff
@@ -43,21 +44,78 @@ This option will specify a list of sources to search from. If a source is not sp
-| Newtonsoft.Json                             | 13.0.3         |         | 3,829,822,911   |
-| Newtonsoft.Json.Bson                        | 1.0.2          |         | 554,641,545     |
-| Newtonsoft.Json.Schema                      | 3.0.15         |         | 39,648,430      |
-| Microsoft.AspNetCore.Mvc.NewtonsoftJson     | 7.0.12         |         | 317,067,823     |
-...
+        Source: nuget.org
+        | Package ID                                  | Latest Version | Authors | Downloads       |
+        |---------------------------------------------|----------------|---------|-----------------|
+        | Newtonsoft.Json                             | 13.0.3         |         | 3,829,822,911   |
+        | Newtonsoft.Json.Bson                        | 1.0.2          |         | 554,641,545     |
+        | Newtonsoft.Json.Schema                      | 3.0.15         |         | 39,648,430      |
+        | Microsoft.AspNetCore.Mvc.NewtonsoftJson     | 7.0.12         |         | 317,067,823     |
+        ...
 
 - Using ``dotnet package search Newtonsoft.Json --exact-match`` on the other side will have the following output
 
-Source: nuget.org
-| Package ID                                  | Latest Version | Authors | Downloads       |
-|---------------------------------------------|----------------|---------|-----------------|
-| Newtonsoft.Json                             | 13.0.3         |         | 3,829,822,911   |
+        Source: nuget.org
+        | Package ID                                  | Latest Version | Authors | Downloads       |
+        |---------------------------------------------|----------------|---------|-----------------|
+        | Newtonsoft.Json                             | 13.0.3         |         | 3,829,822,911   |
+
+#### **Option `--format`**
+
+This option will allow the specification of the output format. The option will take one of these three arguments: Table, List, or Json.
+
+- `--format Table` the output will be in a tabular form as shown below
+
+        Source: nuget.org
+        | Package ID                                  | Latest Version | Authors | Downloads       |
+        |---------------------------------------------|----------------|---------|-----------------|
+        | Newtonsoft.Json                             | 13.0.3         |         | 3,829,822,911   |
+- `--format List` : each package will be listed line by line
+
+        >NuGet.CommandLine | 6.7.0 | Downloads: N/A
+        >NuGet.CommandLine.XPlat | 6.7.0 | Downloads: N/A
+        >NuGet.Commands | 6.7.0 | Downloads: N/A
+        >NuGet.exe | 3.4.3 | Downloads: N/A
+
+- `--format Json` : A json output will be outputted. It will contain the following metdata.
+  - Source
+  - Author
+  - Package ID
+  - LatestVersion
+  - Downloads
+  - Description
+  - Vulnerabilities
+  - Deprecation
+  - Project URL
+
+  The output will take the following format
+
+        [
+            {
+                "source": "source1",
```

Comment:
When reading this PR, it came to my mind that it implies "source name", not "source URL". That's fine. We do utilize source name with our CLI.

Customers shouldn't be getting themselves into this situation, but since it came to my mind.
This is the scenario that I didn't follow the best practice and <clear/> my package sources in my solution's config. My machine-wide config also has nuget.org with a different source name. 

For example, I configured nuget.org under the name "nuget" in solution config and "nugetzz" in machine config.
running `nuget.exe search newtonsoft.json`
I got 2 resultsets: one for nuget and one for nugetzz.

I suppose your command will also do this. Otherwise, you'd have to choose a source name.

I don't have more context to advise you on whether you should group them like
`Source: {{URI}} ({{list of source name(s)}})`
eg, `Source: "https://api.nuget.org/v3/index.json (nuget.org, nugetzz)"`

Reply:
Nice catch Donnie. Currently, my implementation does not group sources. So, it would be 
`"source" : "nuget"`...
`"source" : "nugetzz"`...
However, grouping them by URI is definitely something to think about. I am not sure what the best practice would be for the customers.